### ISSUE TITLE
Proposal: Move to develop+master branch + tags for future releases

### ISSUE THREAD

**Issue by gdevenyi** [BIC-MNI/minc-toolkit-v2#151]
Currently, we have at least two "develop" branches on the go, which have been diverging from master and each other.

I'm doing updates and I'd like to work out a new release, but I'm not sure how I would achieve this right now.

I'd like to propose moving to a develop (PRs against develop, active dev goes there) + master (stable branch, merge develop periodically when "known good"), and tag releases on master.

Thoughts @vfonov? Others?

**Comment by gdevenyi** [comment]
This would also enable a bit more continuous integration (and maybe even auto-builds on release...)

**Comment by vfonov** [comment]
well, I am stuck with switching to ITK-5, too many breaking changes

**Comment by gdevenyi** [comment]
Perhaps that would be "feature branch to be eventually merged into develop"?

Is there a todo/list of things I could potentially help with for the ITK5 porting?

**Comment by bcdarwin** [comment]
What packages remain to be ported -- EZminc and ABC?

**Comment by bcdarwin** [comment]
I agree that the current branching model is leading to stale/unbuildable branches -- potentially could be fixed by having bots look for certain tags and automatically backporting but a simpler model is probably better.

**Comment by bcdarwin** [comment]
These days I already use upstream ANTs/Elastix/c3d (which have moved to ITK5) with no problems (except the usual disagreement between mnc2nii and ITK coordinate system conversion).

**Comment by vfonov** [comment]
i had trouble with elastix , because the version in minc-toolkit contains my modifications to output minc .xfm files. 
Also, I had to disable everything related to diffeomorphic demons.

**Comment by gdevenyi** [comment]
> These days I already use upstream ANTs/Elastix/c3d (which have moved to ITK5) with no problems (except the usual disagreement between mnc2nii and ITK coordinate system conversion).

We Can Rebuilld Him, We Have the Technology

https://github.com/InsightSoftwareConsortium/ITK/pull/147

**Comment by gdevenyi** [comment]
So, can we coordinate on a next step for development?

1. What happens on master branch? Do we merge 1.9.18.2?
2. Can merge 1.9.18-develop into 1.9.19-develop and continue work there?
